Vixen.System/Commands/Named8BitCommand.cs

```diff
@@ -0,0 +1,65 @@
+﻿namespace Vixen.Commands
+{
+    /// <summary>
+    /// Maintains a named 8 bit command.
+    /// </summary>
+    /// <typeparam name="T">Type of index command</typeparam>
+    public class Named8BitCommand<T> : _8BitCommand
+		where T : System.Enum
+	{
+		#region Constructors 
+
+		public Named8BitCommand(byte value) :
+			base(value)
+		{
+		}
+
+		public Named8BitCommand(short value) : 
+			this((byte)value)
+		{
+		}
+
+		public Named8BitCommand(int value) : 
+			this((byte)value)
+		{
+		}
+
+		public Named8BitCommand(long value)
+			: this((byte)value)
+		{
+		}
+
+		public Named8BitCommand(float value)
+			: this((byte)value)
+		{
+		}
+
+		public Named8BitCommand(double value)
+			: this((byte)value)
+		{
+		}
+
+        #endregion
+
+        #region Public Properties
+
+		/// <summary>
+		/// Type of index command value.  This property allows the preview to recognize commands.
+		/// </summary>
+        public T IndexType { get; set; }
+
+		/// <summary>
+		/// Tag associated with the command.
+		/// </summary>
+		/// <remarks>For intelligent fixtures this is the function name.</remarks>
+		public string Tag { get; set; }
+
+		/// <summary>
+		/// Preview representation of the type.
+		/// </summary>
+		/// <remarks>For intelligent fixtures this is helps form a string to indicate what functions are active.</remarks>
+		public string Preview { get; set; }
```

Comment:
Can you give some examples of how this Preview property will be used? Is this just some description type text that is generated when the command is created? Can it be named something more generic than preview? The notion of mentioning preview here just feels wrong. These commands are just data values that any consumer could interpret so we need to avoid pinning them to some particular function.

Reply:
https://www.dropbox.com/s/9xb6wy4uko5j99s/Legend.png?dl=0

This is what the preview looks like with a Gobo and Prism selected.  This property is set to 'G' for the Gobo and 'P' for the prism.  The software is then showing the value associated with that command.

I changed that property name this morning from Legend to Preview but maybe a better name is just label?

Reply:
So the value of that field would be P4, G8 in the example? Descriptive text, but how is it different than Tag which tends to have that meaning in other patterns?

Reply:
The label property is the 'P' and 'G'.  The 4 and the 8 are the DMX command values converted to a string.  The preview iterates over all the commands and checks to see if any have labels and if so creates the P4 G8 strings and concatenates them together and then that legend string is displayed below the fixture.